code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/TokenUtils.java

```diff
@@ -65,25 +65,16 @@ public Account getAccountFromJWT() {
     protected abstract String tokenType();
 
     private Map<String, Object> getJsonData(String jwtKey, String tokenType) {
-        if (StringUtils.isEmpty(jwtKey)) {
-            return null;
-        }
         String toParse;
-        String[] tokenArr = jwtKey.split("\\s+");
-        if (tokenArr.length == 2) {
-            if (!StringUtils.equalsIgnoreCase("bearer", StringUtils.trim(tokenArr[0]))) {
-                return null;
-            }
-            toParse = tokenArr[1];
-        } else if (tokenArr.length == 1) {
-            toParse = tokenArr[0];
-        } else {
-            toParse = jwtKey;
-        }
+        toParse = removeBearer(jwtKey);
```

Comment:
Nit: can we check if jwtKey is empty right before attempting this call? As I can see that inside removeBearer you are checking for an empty value, and then you do the same on line 75